pdc/apps/compose/views.py

```diff
@@ -1399,3 +1402,142 @@ def list(self, request):
             return Response(self._get_composes_for_product_version())
         return Response(status=status.HTTP_400_BAD_REQUEST,
                         data={'detail': 'One of product_version, release or compose argument is required.'})
+
+
+class ComposeTreeViewSet(ChangeSetModelMixin,
+                         StrictQueryParamMixin,
+                         MultiLookupFieldMixin,
+                         viewsets.GenericViewSet):
+    """
+    API endpoint that allows querying compose-variant-arch relevant to location.
+
+    ##Test tools##
+
+    You can use ``curl`` in terminal, with -X _method_ (GET|POST|PUT|DELETE),
+    -d _data_ (a json string). or GUI plugins for
+    browsers, such as ``RESTClient``, ``RESTConsole``.
+    """
+    queryset = ComposeTree.objects.all()
```

Comment:
If some other related objects also need to be accessed, maybe could use select_related() to fetch them with on DB hit.